src/zepben/evolve/model/cim/iec61970/base/auxiliaryequipment/create_auxiliary_equipment_components.py

```diff
@@ -0,0 +1,28 @@
+#  Copyright 2020 Zeppelin Bend Pty Ltd
+#
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+from typing import List
+
+from zepben.evolve import *
+from zepben.evolve.model.common_two_way_connections import add_common_equipment_connections
+
+
+def create_fault_indicator(mrid: str = None, name: str = '', description: str = "", names: List[Name] = None, location: Location = None,
+                           asset_info: AssetInfo = None, in_service: bool = True, normally_in_service: bool = True, usage_points: List[UsagePoint] = None,
+                           equipment_containers: List[EquipmentContainer] = None, operational_restrictions: List[OperationalRestriction] = None,
+                           current_feeders: List[Feeder] = None, terminal: Terminal = None) -> FaultIndicator:
+    """
+    FaultIndicator(AuxiliaryEquipment(Equipment(PowerSystemResource(IdentifiedObject))))
+    IdentifiedObject: mrid, name, description, names
+    PowerSystemResource: location, asset_info
+    Equipment: in_service, normally_in_service, usage_points, equipment_containers, operational_restrictions, current_feeders
+    AuxiliaryEquipment: terminal
+    FaultIndicator:
+    """
+    fi = FaultIndicator(**locals())
+    add_common_equipment_connections(fi, usage_points, equipment_containers, operational_restrictions, current_feeders)
```

Comment:
I like this calling a function to create the reverse links rather than doing in each location. This should be expanded to occur on all classes to prevent missing things.

These function should also be expanded to cover the entire hierachy like the hypothesis field filling and test field verification. i.e. this should be calling `add_common_aux_equipment_connections` , which should be calling `add_common_equipment_connections`. Doing so means you don't accidentally miss things when we modify classes in the hierarchy (e.g. this would miss anything we added to AuxEquipment if it was not updated as part of that change)

All other class constructors should be updated to make a call up the tree for adding these connections

If you do this in conjunction with the suggested change for `UsagePoint` you would only need to be passing the instance to these functions and that will remove a lot of repeated code.